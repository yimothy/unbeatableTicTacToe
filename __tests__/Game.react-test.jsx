import React from 'react';
import renderer from 'react-test-renderer';
import Game from '../client/components/game.jsx';
import Board from '../client/components/board.jsx';
import Square from '../client/components/square.jsx';

// Test components to render correctly
it('Game renders correctly', () => {
  const tree = renderer.create(
    <Game />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Board renders correctly', () => {
  const board = [['X', false, 'O'],
                 [false, 'X', false],
                 ['O', false, 'X']]
  const winningSquares = [[0, 0], [1, 1], [2, 2]];
  const tree = renderer.create(
    <Board board={board} winningSquares={winningSquares} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Square renders correctly', () => {
  const value = 'X';
  const tree = renderer.create(
    <Square value={value} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
