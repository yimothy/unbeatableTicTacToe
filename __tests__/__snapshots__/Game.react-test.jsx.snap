// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders correctly 1`] = `
<div
  style={
    Object {
      "bottom": "0",
      "display": "flex",
      "flexFlow": "row wrap",
      "height": "35vw",
      "left": "0",
      "margin": "auto",
      "minHeight": "310px",
      "minWidth": "310px",
      "right": "0",
      "top": "0",
      "width": "35vw",
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#97d87d",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  >
    X
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  >
    O
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#97d87d",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  >
    X
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  >
    O
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#97d87d",
        "border": "2px solid #dae5bc",
        "float": "left",
        "fontFamily": "Nunito, sans-serif",
        "fontSize": "calc(50px + 4.5vw)",
        "height": "32%",
        "textAlign": "center",
        "width": "32%",
      }
    }
  >
    X
  </div>
</div>
`;

exports[`Game renders correctly 1`] = `
<div
  style={
    Object {
      "fontFamily": "Nunito, sans-serif",
      "width": "100vw",
    }
  }
>
  <div
    style={
      Object {
        "left": "0",
        "margin": "auto",
        "minWidth": "310px",
        "right": "0",
        "width": "35vw",
      }
    }
  >
    <h1>
      IMPOSSIBLE TIC TAC TOE
    </h1>
    <div>
      PLAYING AGAINST: 
      Human
       
      <span>
         | X Turn
      </span>
    </div>
    <div>
      CHOOSE YOUR ENEMY: 
      <span>
         
      </span>
      <button
        onClick={[Function]}
        value="Human"
      >
        HUMAN
      </button>
      <button
        onClick={[Function]}
        value="AI1"
      >
        AI goes first
      </button>
      <button
        onClick={[Function]}
        value="AI2"
      >
        AI goes second
      </button>
    </div>
  </div>
  <div
    style={
      Object {
        "bottom": "0",
        "display": "flex",
        "flexFlow": "row wrap",
        "height": "35vw",
        "left": "0",
        "margin": "auto",
        "minHeight": "310px",
        "minWidth": "310px",
        "right": "0",
        "top": "0",
        "width": "35vw",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "border": "2px solid #dae5bc",
          "float": "left",
          "fontFamily": "Nunito, sans-serif",
          "fontSize": "calc(50px + 4.5vw)",
          "height": "32%",
          "textAlign": "center",
          "width": "32%",
        }
      }
    />
  </div>
</div>
`;

exports[`Square renders correctly 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "border": "2px solid #dae5bc",
      "float": "left",
      "fontFamily": "Nunito, sans-serif",
      "fontSize": "calc(50px + 4.5vw)",
      "height": "32%",
      "textAlign": "center",
      "width": "32%",
    }
  }
>
  X
</div>
`;
